### PR TITLE
Add consul_policy module from dormant PR

### PR DESCRIPTION
**Pull request by valeriopoggi**
##### SUMMARY

Resubmitting parts of #5320 originally from Hakon, as per PR comments. 

Fixes: #1749 

##### ISSUE TYPE
- New Module/Plugin Pull Request

##### COMPONENT NAME
consul_policy

### CODE REVIEW

File: .github/BOTMETA.yml [hunk 1]
```diff
@@ -1394,7 +1394,7 @@ macros:
   team_ansible_core:
   team_aix: MorrisA bcoca d-little flynn1973 gforster kairoaraujo marvin-sinister mator molekuul ramooncamacho wtcross
   team_bsd: JoergFiedler MacLemon bcoca dch jasperla mekanix opoplawski overhacked tuxillo
-  team_consul: sgargan
+  team_consul: sgargan Hakon
```
Comment (felixfontein) [inline .github/BOTMETA.yml:1397]: This volunteers Hakon as a maintainer for the consul modules. @Hakon do you agree to this?

File: plugins/modules/consul_policy.py [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: consul_policy
+short_description: Manipulate Consul policies
+version_added: 8.1.0
```
Comment (felixfontein) [inline plugins/modules/consul_policy.py:14]:
```suggestion
version_added: 7.2.0
```

File: plugins/modules/consul_policy.py [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    default: http
+    type: str
+  token:
+    description:
+      - A management token is required to manipulate the policies.
+    type: str
+  validate_certs:
+    type: bool
+    description:
+      - Whether to verify the tls certificate of the consul agent or not.
```
Comment (felixfontein) [inline plugins/modules/consul_policy.py:77]:
```suggestion
      - Whether to verify the TLS certificate of the consul agent or not.
```

File: plugins/modules/consul_policy.py [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+    required: false
+    default: True
```
Comment (felixfontein) [inline plugins/modules/consul_policy.py:79]:
```suggestion
    default: true
```

File: plugins/modules/consul_policy.py [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+  community.general.consul_policy:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-access
+    state: absent
+"""
+
+RETURN = """
+operation:
+    description: the operation performed on the policy
```
Comment (felixfontein) [inline plugins/modules/consul_policy.py:124]:
```suggestion
    description: The operation performed on the policy.
```

File: plugins/modules/consul_policy.py [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+DOCUMENTATION = '''
+module: consul_policy
+short_description: Manipulate Consul policies
+version_added: 8.1.0
+description:
+ - Allows the addition, modification and deletion of policies in a consul
+   cluster via the agent. For more details on using and configuring ACLs,
+   see U(https://www.consul.io/docs/guides/acl.html).
+author:
+  - Håkon Lerring (@Hakon)
```
Comment (felixfontein) [inline plugins/modules/consul_policy.py:20]:
```suggestion
  - Håkon Lerring (@Hakon)
extends_documentation_fragment:
  - community.general.attributes
attributes:
  check_mode:
    support: none
  diff_mode:
    support: none
```

File: plugins/modules/consul_session.py [hunk 7]
```diff
@@ -20,36 +20,30 @@
   - python-consul
   - requests
 author:
-  - Steve Gargan (@sgargan)
-extends_documentation_fragment:
```
Comment (felixfontein) [inline plugins/modules/consul_session.py:24]: Please undo removing all documentation improvements from this module.

Also, better move changes in this module to a separate PR.

File: tests/integration/targets/consul/tasks/consul_session.yml [hunk 8]
```diff
@@ -45,7 +48,7 @@
 
 - name: search created session
   set_fact:
-    test_session_found: true
+    test_session_found: True
```
Comment (felixfontein) [inline tests/integration/targets/consul/tasks/consul_session.yml:51]: YAML booleans must be lower-case.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/modules/consul_session.py (hunk 7 above)
Comment (valeriopoggi) [reply plugins/modules/consul_session.py:24]: Thanks for reviewing! Sure, I'll split this into a separate PR

File: .github/BOTMETA.yml (hunk 1 above)
Comment (valeriopoggi) [reply .github/BOTMETA.yml:1397]: @Hakon put in all the hard work here, I wasn't sure about what to do here. 

I'm guessing if we don't get a reply or confirmation, I can revert this change.

Comment (valeriopoggi) [reply .github/BOTMETA.yml]: I marked  @Hakon as inactive maintainer in #6755. 

This PR depends on code in the other PR, so leaving this as draft until the other one gets merged to avoid having to fix merge conflicts.

**Comment by felixfontein** [comment]
Please rebase now that #6755 has been merged so that the changes from that PR no longer show up under "Files changed".
